socks.py

```diff
@@ -834,29 +834,49 @@ def connect(self, dest_pair):
         except socket.error as error:
             # Error while connecting to proxy
             self.close()
-            proxy_addr, proxy_port = proxy_addr
-            proxy_server = "{0}:{1}".format(proxy_addr, proxy_port)
-            printable_type = PRINTABLE_PROXY_TYPES[proxy_type]
-
-            msg = "Error connecting to {0} proxy {1}".format(printable_type,
-                                                             proxy_server)
-            log.debug("%s due to: %s", msg, error)
-            raise ProxyConnectionError(msg, error)
+            if not catch_errors:
+                    proxy_addr, proxy_port = proxy_addr
```

Comment:
This block and other blocks are indented with 8 spaces, but the existing code is indented with 4 spaces.